### DISCUSSION TITLE
A few questions

### DISCUSSION THREAD

**Discussion by gurupras** [wegylexy/rnnoise_wasm#15] [Q&A]
Hello,
This has been an incredibly useful repo in getting started with rnnoise and `AudioWorkletProcessor`. Thank you for this work.

I did have some questions though, and was hoping you'd be able to help me understand this a little better.

**Does worklet.c _need_ to be in C?**
I was able to make a JavaScript equivalent of this for the most part and was wondering if it really needed to be in C. Performance reasons perhaps?

**What is getInput really returning?**
While making a JS version of worklet.c, the return value of `getInput` stumped me. It looks to me like it is returning a pointer, yet somehow JS is performing arithmetic on the returned value (`instance.getInput(this.state) / 4`). Could you please explain what is happening here?

**Comment by wegylexy** [comment]
If you know C well, 4 is `sizeof(float)`. The source language does not matter, but the WebAssembly runtime is likely better optimized than the engine for vanilla JavaScript. However, the primary reason is that the original cross-platform RNNoise source repo is maintained in C by a group of developers. If you port it to another language, you may be on your own.

My implementation is merely a wrapper around RNNoise for use in the browser.
